docs/arch/adr-40.md

```diff
@@ -0,0 +1,25 @@
+# ADR 40: Next.js Internationalized Routing
+
+28 December 2021
+
+## Context
+
+Our goal is to internationalize FEM and provide translatable content based on a user's preferred locale. There are two types of translatable strings: static strings (e.g menu items, nav links) and dynamic strings (project-specific content provided by volunteers).
+
+When a user specifies their locale preference, all translatable content should reflect that preference across the Zooniverse platform.
+
+## Decision
+
+In order to synchronize a user's locale preference across Zooniverse, we'll utilize [Next.js Internationalized Sub-path Routing](https://nextjs.org/docs/advanced-features/i18n-routing#sub-path-routing). With the exception of English (the default locale), the user's chosen locale is reflected in a url subpath.
+
+For example, `www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess` will render content in English because `en` is the default locale. When a user decides to view the same page in French, the url will change to `www.zooniverse.org/projects/fr/nora-dot-eisner/planet-hunters-tess`. As explained in [ADR #36](https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-36.md), `/projects` is the base path for `app-project`, so `/fr/` falls after `/projects`. Therefore, the url to view Zooniverse About pages in French will look like `www.zooniverse.org/about/fr/publications`.
```

Comment:
Noting we may have to update our CDN / app routing rules to accommodate the extra language path to correctly route the traffic. I'm happy to help with this. 

These are the current routing rules that proxy traffic to the correct backends https://github.com/zooniverse/static/blob/3760ac0848fc6e8289e55663d7bb930a249d4325/nginx-fem-redirects.conf

and they are included on the www.zooniverse.org domain via this https://github.com/zooniverse/static/blob/3760ac0848fc6e8289e55663d7bb930a249d4325/sites/www.zooniverse.org.conf#L3

Reply:
If we could publish URLs as eg. `www.zooniverse.org/about/publications?language=fr`, then use middleware to map that query param to `www.zooniverse.org/about/fr/publications` then we won't need to update our CDN config or the router config in PFE.

I'm not sure what the implications would be for link URLs published by the `next/link` component. It might be we don't have much control over the published URL there.

Reply:
that sounds like a good idea - i was kind of assuming URL query params would be a natural fit for translation strings and backwards compatible with the PFE service. 

We would most likely have to add query param caching to the CDN service on www.zooniverse.org but that's a simple fix.